{
  "affected": [
    {
      "database_specific": {
        "source": "https://storage.googleapis.com/cve-osv-conversion/osv-output/CVE-2021-29566.json",
        "vanir_signatures": [
          {
            "deprecated": false,
            "digest": {
              "line_hashes": [
                "96563779246992782287516591343516453836",
                "260791232739742738662922402804201275629",
                "240656298568355502289163248359206752419",
                "310252771468143506574772725476797846361",
                "96563779246992782287516591343516453836",
                "260791232739742738662922402804201275629",
                "240656298568355502289163248359206752419",
                "225767627981085726337480589104864560893",
                "44097201705485675186745246342504614398",
                "171162292111960105019969332131039616978",
                "74932482180298867081423578408567783784",
                "250351533755703723792816919320122242261",
                "32078003309922402579923077891902987930",
                "96563779246992782287516591343516453836",
                "260791232739742738662922402804201275629",
                "240656298568355502289163248359206752419",
                "225767627981085726337480589104864560893",
                "316863186184370311580529875121008744043",
                "315361678216214071101843588402153035479",
                "190417965879306103159391078031537172778",
                "85308760875451375968903868289181003895",
                "32078003309922402579923077891902987930"
              ],
              "threshold": 0.9
            },
            "id": "CVE-2021-29566-ca4057d6",
            "signature_type": "Line",
            "signature_version": "v1",
            "source": "https://github.com/tensorflow/tensorflow/commit/3f6fe4dfef6f57e768260b48166c27d148f3015f",
            "target": {
              "file": "tensorflow/core/kernels/dilation_ops.cc"
            }
          }
        ]
      },
      "ranges": [
        {
          "events": [
            {
              "introduced": "0"
            },
            {
              "fixed": "3f6fe4dfef6f57e768260b48166c27d148f3015f"
            }
          ],
          "repo": "https://github.com/tensorflow/tensorflow",
          "type": "GIT"
        }
      ],
      "versions": [
        "0.12.0-rc0",
        "0.12.0-rc1",
        "0.12.1",
        "0.5.0",
        "0.6.0",
        "v0.10.0",
        "v0.10.0rc0",
        "v0.11.0",
        "v0.11.0rc0",
        "v0.11.0rc1",
        "v0.11.0rc2",
        "v0.12.0",
        "v0.7.0",
        "v0.7.1",
        "v0.8.0rc0",
        "v0.9.0",
        "v0.9.0rc0",
        "v1.0.0",
        "v1.0.0-alpha",
        "v1.0.0-rc0",
        "v1.0.0-rc1",
        "v1.0.0-rc2",
        "v1.1.0",
        "v1.1.0-rc0",
        "v1.1.0-rc1",
        "v1.1.0-rc2",
        "v1.12.0",
        "v1.12.0-rc0",
        "v1.12.0-rc1",
        "v1.12.0-rc2",
        "v1.12.1",
        "v1.2.0",
        "v1.2.0-rc0",
        "v1.2.0-rc1",
        "v1.2.0-rc2",
        "v1.3.0-rc0",
        "v1.3.0-rc1",
        "v1.5.0",
        "v1.5.0-rc0",
        "v1.5.0-rc1",
        "v1.6.0",
        "v1.6.0-rc0",
        "v1.6.0-rc1",
        "v1.7.0",
        "v1.7.0-rc0",
        "v1.7.0-rc1",
        "v1.8.0",
        "v1.8.0-rc0",
        "v1.8.0-rc1",
        "v1.9.0",
        "v1.9.0-rc0",
        "v1.9.0-rc1",
        "v1.9.0-rc2"
      ]
    }
  ],
  "aliases": [
    "BIT-tensorflow-2021-29566",
    "GHSA-pvrc-hg3f-58r6",
    "PYSEC-2021-203",
    "PYSEC-2021-494",
    "PYSEC-2021-692"
  ],
  "details": "TensorFlow is an end-to-end open source platform for machine learning. An attacker can write outside the bounds of heap allocated arrays by passing invalid arguments to `tf.raw_ops.Dilation2DBackpropInput`. This is because the implementation(https://github.com/tensorflow/tensorflow/blob/afd954e65f15aea4d438d0a219136fc4a63a573d/tensorflow/core/kernels/dilation_ops.cc#L321-L322) does not validate before writing to the output array. The values for `h_out` and `w_out` are guaranteed to be in range for `out_backprop` (as they are loop indices bounded by the size of the array). However, there are no similar guarantees relating `h_in_max`/`w_in_max` and `in_backprop`. The fix will be included in TensorFlow 2.5.0. We will also cherrypick this commit on TensorFlow 2.4.2, TensorFlow 2.3.3, TensorFlow 2.2.3 and TensorFlow 2.1.4, as these are also affected and still in supported range.",
  "id": "CVE-2021-29566",
  "modified": "2025-10-14T18:36:12.923514Z",
  "published": "2021-05-14T20:15:13Z",
  "references": [
    {
      "type": "FIX",
      "url": "https://github.com/tensorflow/tensorflow/commit/3f6fe4dfef6f57e768260b48166c27d148f3015f"
    },
    {
      "type": "FIX",
      "url": "https://github.com/tensorflow/tensorflow/security/advisories/GHSA-pvrc-hg3f-58r6"
    }
  ],
  "related": [
    "GHSA-pvrc-hg3f-58r6"
  ],
  "schema_version": "1.7.3",
  "severity": [
    {
      "score": "CVSS:3.1/AV:L/AC:L/PR:L/UI:N/S:U/C:H/I:H/A:H",
      "type": "CVSS_V3"
    }
  ]
}